pkg/oc/admin/router/router.go

```diff
@@ -229,6 +235,23 @@ type RouterConfig struct {
 	StrictSNI bool
 
 	Local bool
+
+	// MutualTLSAuth controls access to the router using a mutually agreed
+	// upon TLS authentication mechanism (example client certificates).
+	// One of: required | optional | none  - the default is none.
+	MutualTLSAuth string
+
+	// MutualTLSAuthCA contains the CA certificates that will be used
+	// to verify a client's certificate.
+	MutualTLSAuthCA string
+
+	// MutualTLSAuthCRL contains the certificate revocation list used to
+	// verify a client's certificate.
+	MutualTLSAuthCRL string
+
+	// MutualTLSAuthFilter contains the value to filter requests based on
+	// a client certificate subject field substring match.
+	MutualTLSAuthFilter string
```

Comment:
Won't this match also: /CN=blah/CN=blah/CN=more-blah/CN=header.test/ST=CA/C=US/O=Security/OU=OpenShift3 ?

Reply:
Yes, it would string substring match that.  But that said, is that a valid DN/subject? 
From the looks of it the / is a separator in the subject, correct? 
So shouldn't that be escaped if someone forces that into an input field? Otherwise, there's potential of other issues, no?

Reply:
@simo5  also pertinent here is the fact that the `ROUTER_MUTUAL_TLS_AUTH_FILTER` is optional. Meaning if its not specified, we don't do any restrictive checks.  This is a secondary check. 

But if this is still a sticking point for you, we could potentially change the substring match to a regular expression ... so you could potentially do a `^<escaped-text>$` kind of expression for an exact match.
That said, it does make the user interface a bit more awkward as it requires the user to specify a regular expression.  Not a back breaker though!
And it is worth noting that a regular expression without the positional anchor ... ala `CN=foo` would still match a `/CN=blah/CN=foo/CN=blah/` string.

@knobunc / @ironcladlou  any issues with that?

Reply:
You can have as many CN elements as you want in a DN.
Changing it to a regular expression would be definitely better and I would accept as a solution because then you can do exact matches if you want or anything.

An example with exact match and one with subfiltering should be provided as doc then.